src/otx/engine/engine.py

```diff
@@ -346,7 +348,7 @@ def predict(
         return self.trainer.predict(
             model=lit_module,
             datamodule=datamodule if datamodule is not None else self.datamodule,
-            ckpt_path=str(checkpoint) if checkpoint is not None else self.checkpoint,
+            ckpt_path=str(checkpoint) if checkpoint is not None else str(self.checkpoint),
```

Comment:
Won't this cause problems if both `checkpoint` and `self.checkpoint` are `None`?

Reply:
Also curious about why we need this change

Reply:
It's just to satisfy type checker, but I can revert this. `trainer.predict` takes a string (by annotation), I don't know what happens if we pass a Path object. Usually, `checkpoint` is not None, so we just don't hit the second part of if. I think it'd be a file reading failure.